### PR TITLE
IMBU-38 multiple models

### PR DESCRIPTION
**Pull request by oxtopus**
@lscheinkman This should do the trick.  `GET /fluent`, and it'll return "true" (models have been created and are ready to handle queries) or "false" (models are still being created and trained and are NOT ready to handle queries).

In my experience, it takes a few minutes after first startup for the endpoint to start returning "true".

### CODE REVIEW

**Comment by oxtopus** [comment]
Though, FWIW, after merging master the docker build fails for me:

```
...
Step 15 : RUN npm run build
 ---> Running in 905f2a6cf6f1

> Imbu@0.0.1 prebuild /opt/numenta/imbu
> npm run lint


> Imbu@0.0.1 lint /opt/numenta/imbu
> eslint --ext=js --ext=jsx gui/ tests/js/

sh: 1: eslint: not found

npm ERR! Linux 4.1.13-boot2docker
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "lint"
npm ERR! node v4.2.3
npm ERR! npm  v2.14.7
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! Imbu@0.0.1 lint: `eslint --ext=js --ext=jsx gui/ tests/js/`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the Imbu@0.0.1 lint script 'eslint --ext=js --ext=jsx gui/ tests/js/'.
npm ERR! This is most likely a problem with the Imbu package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     eslint --ext=js --ext=jsx gui/ tests/js/
npm ERR! You can get their info via:
npm ERR!     npm owner ls Imbu
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /opt/numenta/imbu/npm-debug.log

npm ERR! Linux 4.1.13-boot2docker
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "build"
npm ERR! node v4.2.3
npm ERR! npm  v2.14.7
npm ERR! code ELIFECYCLE
npm ERR! Imbu@0.0.1 prebuild: `npm run lint`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the Imbu@0.0.1 prebuild script 'npm run lint'.
npm ERR! This is most likely a problem with the Imbu package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run lint
npm ERR! You can get their info via:
npm ERR!     npm owner ls Imbu
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /opt/numenta/imbu/npm-debug.log
The command '/bin/sh -c npm run build' returned a non-zero code: 1
```

**Comment by lscheinkman** [comment]
Thanks, I will take a look at the docker issue.

**Comment by lscheinkman** [comment]
@oxtopus Docker built for me from your branch. Can you check if `eslint` was installed in `Step 13`?
Here is my log:
```
...
Step 13 : RUN npm install
 ---> Running in 23e088014c0c
npm WARN package.json Imbu@0.0.1 No README data
npm WARN deprecated babel-core@5.8.23: Babel 5 is no longer being maintained. Upgrade to Babel 6.
npm WARN optional dep failed, continuing fsevents@1.0.6
npm WARN optional dep failed, continuing fsevents@1.0.6

> caniuse-api@1.4.1 postinstall /opt/numenta/imbu/node_modules/material-ui/node_modules/inline-style-prefixer/node_modules/caniuse-api
> node generator.js

caniuse-api: Generation ok
react-addons-create-fragment@0.14.3 node_modules/react-addons-create-fragment

react-addons-update@0.14.3 node_modules/react-addons-update

react-addons-transition-group@0.14.3 node_modules/react-addons-transition-group

react-addons-pure-render-mixin@0.14.3 node_modules/react-addons-pure-render-mixin

react-dom@0.14.3 node_modules/react-dom

eslint-plugin-react@3.10.0 node_modules/eslint-plugin-react

fluxible-addons-react@0.2.0 node_modules/fluxible-addons-react
└── hoist-non-react-statics@1.0.3

fluxible@1.0.3 node_modules/fluxible
├── setimmediate@1.0.4
├── is-promise@2.1.0
├── debug@2.2.0 (ms@0.7.1)
└── dispatchr@0.3.3 (inherits@2.0.1, eventemitter3@1.1.1)

superagent@1.3.0 node_modules/superagent
├── extend@1.2.1
├── cookiejar@2.0.1
├── methods@1.0.1
├── reduce-component@1.0.1
├── component-emitter@1.1.2
├── mime@1.3.4
├── qs@2.3.3
├── debug@2.2.0 (ms@0.7.1)
├── readable-stream@1.0.27-1 (string_decoder@0.10.31, isarray@0.0.1, inherits@2.0.1, core-util-is@1.0.2)
├── formidable@1.0.14
└── form-data@0.2.0 (async@0.9.2, mime-types@2.0.14, combined-stream@0.0.7)

webpack-stream@2.1.1 node_modules/webpack-stream
├── memory-fs@0.2.0
├── through@2.3.8
├── vinyl@0.5.3 (clone-stats@0.0.1, replace-ext@0.0.1, clone@1.0.2)
└── lodash.clone@3.0.3 (lodash._isiterateecall@3.0.9, lodash._bindcallback@3.0.1, lodash._baseclone@3.3.0)

babel-eslint@4.1.3 node_modules/babel-eslint
├── acorn-to-esprima@1.0.7
├── lodash.assign@3.2.0 (lodash._baseassign@3.2.0, lodash.keys@3.1.2, lodash._createassigner@3.1.1)
└── lodash.pick@3.1.0 (lodash.restparam@3.6.1, lodash._pickbyarray@3.0.2, lodash._bindcallback@3.0.1, lodash._baseflatten@3.1.4, lodash._pickbycallback@3.0.0)

mocha@2.3.4 node_modules/mocha
├── escape-string-regexp@1.0.2
├── commander@2.3.0
├── diff@1.4.0
├── supports-color@1.2.0
├── growl@1.8.1
├── debug@2.2.0 (ms@0.7.1)
├── mkdirp@0.5.0 (minimist@0.0.8)
├── glob@3.2.3 (inherits@2.0.1, graceful-fs@2.0.3, minimatch@0.2.14)
└── jade@0.26.3 (commander@0.6.1, mkdirp@0.3.0)

babel-plugin-typecheck@3.1.0 node_modules/babel-plugin-typecheck

gulp@3.9.0 node_modules/gulp
├── interpret@0.6.6
├── pretty-hrtime@1.0.1
├── deprecated@0.0.1
├── archy@1.0.0
├── minimist@1.2.0
├── semver@4.3.6
├── tildify@1.1.2 (os-homedir@1.0.1)
├── v8flags@2.0.10 (user-home@1.1.1)
├── chalk@1.1.1 (escape-string-regexp@1.0.3, ansi-styles@2.1.0, supports-color@2.0.0, strip-ansi@3.0.0, has-ansi@2.0.0)
├── orchestrator@0.3.7 (sequencify@0.0.7, stream-consume@0.1.0, end-of-stream@0.1.5)
├── liftoff@2.2.0 (extend@2.0.1, rechoir@0.6.2, flagged-respawn@0.3.1, resolve@1.1.6, findup-sync@0.3.0)
└── vinyl-fs@0.3.14 (graceful-fs@3.0.8, defaults@1.0.3, strip-bom@1.0.0, vinyl@0.4.6, through2@0.6.5, mkdirp@0.5.1, glob-stream@3.1.18, glob-watcher@0.0.6)

babel-loader@5.3.2 node_modules/babel-loader
├── object-assign@3.0.0
└── loader-utils@0.2.12 (big.js@3.1.3, json5@0.4.0)

gulp-util@3.0.7 node_modules/gulp-util
├── array-differ@1.0.0
├── array-uniq@1.0.2
├── lodash._reevaluate@3.0.0
├── fancy-log@1.1.0
├── lodash._reescape@3.0.0
├── object-assign@3.0.0
├── lodash._reinterpolate@3.0.0
├── beeper@1.1.0
├── replace-ext@0.0.1
├── minimist@1.2.0
├── has-gulplog@0.1.0 (sparkles@1.0.0)
├── vinyl@0.5.3 (clone-stats@0.0.1, clone@1.0.2)
├── gulplog@1.0.0 (glogg@1.0.0)
├── lodash.template@3.6.2 (lodash._isiterateecall@3.0.9, lodash.restparam@3.6.1, lodash._basecopy@3.0.1, lodash._basetostring@3.0.1, lodash._basevalues@3.0.0, lodash.escape@3.0.0, lodash.templatesettings@3.1.0, lodash.keys@3.1.2)
├── chalk@1.1.1 (escape-string-regexp@1.0.3, ansi-styles@2.1.0, supports-color@2.0.0, strip-ansi@3.0.0, has-ansi@2.0.0)
├── through2@2.0.0 (xtend@4.0.1, readable-stream@2.0.4)
├── multipipe@0.1.2 (duplexer2@0.0.2)
└── dateformat@1.0.12 (get-stdin@4.0.1, meow@3.6.0)

webpack@1.12.9 node_modules/webpack
├── interpret@0.6.6
├── tapable@0.1.10
├── clone@1.0.2
├── async@1.5.0
├── esprima@2.7.1
├── supports-color@3.1.2 (has-flag@1.0.0)
├── enhanced-resolve@0.9.1 (graceful-fs@4.1.2, memory-fs@0.2.0)
├── mkdirp@0.5.1 (minimist@0.0.8)
├── optimist@0.6.1 (wordwrap@0.0.3, minimist@0.0.10)
├── memory-fs@0.3.0 (errno@0.1.4, readable-stream@2.0.4)
├── webpack-core@0.6.8 (source-map@0.4.4, source-list-map@0.1.5)
├── loader-utils@0.2.12 (big.js@3.1.3, json5@0.4.0)
├── watchpack@0.2.9 (graceful-fs@4.1.2, async@0.9.2, chokidar@1.4.1)
├── uglify-js@2.6.1 (async@0.2.10, uglify-to-browserify@1.0.2, source-map@0.5.3, yargs@3.10.0)
└── node-libs-browser@0.5.3 (string_decoder@0.10.31, https-browserify@0.0.0, tty-browserify@0.0.0, path-browserify@0.0.0, constants-browserify@0.0.1, punycode@1.3.2, os-browserify@0.1.2, process@0.11.2, assert@1.3.0, querystring-es3@0.2.1, timers-browserify@1.4.2, stream-browserify@1.0.0, events@1.1.0, readable-stream@1.1.13, vm-browserify@0.0.4, util@0.10.3, console-browserify@1.1.0, url@0.10.3, http-browserify@1.7.0, buffer@3.5.4, browserify-zlib@0.1.4, domain-browser@1.1.5, crypto-browserify@3.2.8)

babel@5.8.23 node_modules/babel
├── path-is-absolute@1.0.0
├── slash@1.0.0
├── path-exists@1.0.0
├── fs-readdir-recursive@0.1.2
├── convert-source-map@1.1.2
├── commander@2.9.0 (graceful-readlink@1.0.1)
├── source-map@0.4.4 (amdefine@1.0.0)
├── glob@5.0.15 (inherits@2.0.1, once@1.3.3, inflight@1.0.4, minimatch@3.0.0)
├── output-file-sync@1.1.1 (xtend@4.0.1, mkdirp@0.5.1)
├── chokidar@1.4.1 (inherits@2.0.1, glob-parent@2.0.0, async-each@0.1.6, is-binary-path@1.0.1, is-glob@2.0.1, readdirp@2.0.0, anymatch@1.3.0)
└── lodash@3.10.1

esdoc@0.4.3 node_modules/esdoc
├── escape-html@1.0.2
├── estraverse@4.1.0
├── ice-cap@0.0.3
├── color-logger@0.0.3
├── taffydb@2.7.2
├── marked@0.3.5
├── minimist@1.1.1
├── espree@2.2.0
├── escodegen@1.7.0 (estraverse@1.9.3, esutils@2.0.2, esprima@1.2.5, optionator@0.5.0, source-map@0.2.0)
├── fs-extra@0.20.1 (jsonfile@2.2.3, graceful-fs@3.0.8, rimraf@2.4.4)
├── cheerio@0.19.0 (entities@1.1.1, dom-serializer@0.1.0, css-select@1.0.0, htmlparser2@3.8.3, lodash@3.10.1)
└── core-js@0.9.18

esdoc-es7-plugin@0.0.3 node_modules/esdoc-es7-plugin
├── estraverse@4.1.0
├── babylon@5.8.20
└── core-js@1.0.1

esformatter@0.7.3 node_modules/esformatter
├── rocambole-node@1.0.0
├── stdin@0.0.1
├── debug@0.7.4
├── strip-json-comments@0.1.3
├── supports-color@1.3.1
├── rocambole-token@1.2.1
├── user-home@2.0.0 (os-homedir@1.0.1)
├── minimist@1.2.0
├── semver@2.2.1
├── espree@1.12.3
├── disparity@2.0.0 (diff@1.4.0, ansi-styles@2.1.0)
├── rocambole@0.7.0 (esprima@2.7.1)
├── rocambole-whitespace@1.0.0 (repeat-string@1.5.2, debug@2.2.0)
├── rocambole-linebreak@1.0.1 (semver@4.3.6, debug@2.2.0)
├── rocambole-indent@2.0.4 (debug@2.2.0)
├── glob@5.0.15 (path-is-absolute@1.0.0, inherits@2.0.1, once@1.3.3, inflight@1.0.4, minimatch@3.0.0)
├── resolve@1.1.6
├── npm-run@1.1.1 (sync-exec@0.5.0, serializerr@1.0.2, npm-path@1.0.2)
└── mout@0.11.1

react-tap-event-plugin@0.2.1 node_modules/react-tap-event-plugin
└── fbjs@0.2.1 (whatwg-fetch@0.9.0, promise@7.0.4, core-js@1.2.6)

eslint@1.10.1 node_modules/eslint
├── escape-string-regexp@1.0.3
├── path-is-absolute@1.0.0
├── object-assign@4.0.1
├── path-is-inside@1.0.1
├── globals@8.14.0
├── xml-escape@1.0.0
├── esutils@2.0.2
├── strip-json-comments@1.0.4
├── estraverse@4.1.1
├── estraverse-fb@1.3.1
├── text-table@0.2.0
├── user-home@2.0.0 (os-homedir@1.0.1)
├── espree@2.2.5
├── is-resolvable@1.0.0 (tryit@1.0.2)
├── debug@2.2.0 (ms@0.7.1)
├── optionator@0.6.0 (fast-levenshtein@1.0.7, type-check@0.3.1, levn@0.2.5, wordwrap@0.0.3, deep-is@0.1.3, prelude-ls@1.1.2)
├── doctrine@0.7.2 (isarray@0.0.1, esutils@1.1.6)
├── chalk@1.1.1 (supports-color@2.0.0, ansi-styles@2.1.0, has-ansi@2.0.0, strip-ansi@3.0.0)
├── glob@5.0.15 (inherits@2.0.1, once@1.3.3, inflight@1.0.4)
├── json-stable-stringify@1.0.0 (jsonify@0.0.0)
├── lodash.merge@3.3.2 (lodash._arrayeach@3.0.0, lodash._arraycopy@3.0.0, lodash._getnative@3.9.1, lodash.keysin@3.0.8, lodash.keys@3.1.2, lodash.isarguments@3.0.4, lodash.isarray@3.0.4, lodash.istypedarray@3.0.2, lodash.isplainobject@3.2.0, lodash._createassigner@3.1.1, lodash.toplainobject@3.0.0)
├── minimatch@3.0.0 (brace-expansion@1.1.2)
├── shelljs@0.5.3
├── mkdirp@0.5.1 (minimist@0.0.8)
├── lodash.clonedeep@3.0.2 (lodash._bindcallback@3.0.1, lodash._baseclone@3.3.0)
├── concat-stream@1.5.1 (inherits@2.0.1, typedarray@0.0.6, readable-stream@2.0.4)
├── lodash.omit@3.1.0 (lodash._bindcallback@3.0.1, lodash._pickbyarray@3.0.2, lodash.restparam@3.6.1, lodash._arraymap@3.0.0, lodash._pickbycallback@3.0.0, lodash._baseflatten@3.1.4, lodash.keysin@3.0.8, lodash._basedifference@3.0.3)
├── is-my-json-valid@2.12.3 (jsonpointer@2.0.0, generate-function@2.0.0, xtend@4.0.1, generate-object-property@1.2.0)
├── file-entry-cache@1.2.4 (flat-cache@1.0.10)
├── handlebars@4.0.5 (async@1.5.0, source-map@0.4.4, optimist@0.6.1, uglify-js@2.6.1)
├── inquirer@0.11.0 (ansi-regex@2.0.0, strip-ansi@3.0.0, ansi-escapes@1.1.0, figures@1.4.0, cli-width@1.1.0, through@2.3.8, rx-lite@3.1.2, run-async@0.1.0, readline2@1.0.1, cli-cursor@1.0.2, lodash@3.10.1)
├── js-yaml@3.4.6 (inherit@2.2.2, esprima@2.7.1, argparse@1.0.3)
└── escope@3.3.0 (esrecurse@3.1.1, es6-weak-map@2.0.1, es6-map@0.1.3)

babel-core@5.8.23 node_modules/babel-core
├── slash@1.0.0
├── path-exists@1.0.0
├── fs-readdir-recursive@0.1.2
├── path-is-absolute@1.0.0
├── try-resolve@1.0.1
├── babel-plugin-remove-console@1.0.1
├── babel-plugin-remove-debugger@1.0.1
├── babel-plugin-react-display-name@1.0.3
├── babel-plugin-property-literals@1.0.1
├── babel-plugin-react-constant-elements@1.0.3
├── babel-plugin-constant-folding@1.0.1
├── babel-plugin-eval@1.0.1
├── babel-plugin-jscript@1.0.4
├── babel-plugin-inline-environment-variables@1.0.1
├── babel-plugin-dead-code-elimination@1.0.2
├── babel-plugin-undefined-to-void@1.1.6
├── babel-plugin-member-expression-literals@1.0.1
├── shebang-regex@1.0.0
├── to-fast-properties@1.0.1
├── trim-right@1.0.1
├── babel-plugin-proto-to-assign@1.0.4
├── babel-plugin-runtime@1.0.7
├── globals@6.4.1
├── esutils@2.0.2
├── private@0.1.6
├── convert-source-map@1.1.2
├── js-tokens@1.0.1
├── home-or-tmp@1.0.0 (os-tmpdir@1.0.1, user-home@1.1.1)
├── debug@2.2.0 (ms@0.7.1)
├── babel-plugin-undeclared-variables-check@1.0.2 (leven@1.0.2)
├── line-numbers@0.2.0 (left-pad@0.0.3)
├── babylon@5.8.34
├── chalk@1.1.1 (escape-string-regexp@1.0.3, ansi-styles@2.1.0, supports-color@2.0.0, has-ansi@2.0.0, strip-ansi@3.0.0)
├── repeating@1.1.3 (is-finite@1.0.1)
├── is-integer@1.0.6 (is-finite@1.0.1)
├── source-map@0.4.4 (amdefine@1.0.0)
├── minimatch@2.0.10 (brace-expansion@1.1.2)
├── detect-indent@3.0.1 (get-stdin@4.0.1, minimist@1.2.0)
├── bluebird@2.10.2
├── output-file-sync@1.1.1 (xtend@4.0.1, mkdirp@0.5.1)
├── resolve@1.1.6
├── source-map-support@0.2.10 (source-map@0.1.32)
├── regexpu@1.3.0 (regjsgen@0.2.0, regenerate@1.2.1, esprima@2.7.1, regjsparser@0.1.5, recast@0.10.39)
├── json5@0.4.0
├── regenerator@0.8.35 (through@2.3.8, esprima-fb@15001.1.0-dev-harmony-fb, recast@0.10.24, commoner@0.10.4, defs@1.1.1)
├── lodash@3.10.1
└── core-js@1.2.6

react@0.14.3 node_modules/react
├── envify@3.4.0 (through@2.3.8, jstransform@10.1.0)
└── fbjs@0.3.2 (whatwg-fetch@0.9.0, ua-parser-js@0.7.9, loose-envify@1.1.0, promise@7.0.4, core-js@1.2.6)

material-ui@0.13.4 node_modules/material-ui
├── lodash.throttle@3.0.4
├── warning@2.1.0 (loose-envify@1.1.0)
├── lodash.debounce@3.1.1 (lodash._getnative@3.9.1)
└── inline-style-prefixer@0.5.4 (bowser@1.0.0, caniuse-api@1.4.1)
 ---> 1c16a5f189a9
Removing intermediate container 23e088014c0c
Step 14 : COPY . /opt/numenta/imbu
 ---> 7bf91158b2a9
Removing intermediate container 1e6a38ac755f
Step 15 : RUN npm run build
 ---> Running in 0af939449549

> Imbu@0.0.1 prebuild /opt/numenta/imbu
> npm run lint


> Imbu@0.0.1 lint /opt/numenta/imbu
> eslint --ext=js --ext=jsx gui/ tests/js/


/opt/numenta/imbu/gui/browser/actions/search-query.js
  27:14  warning  Variables within the same declaration block should be sorted alphabetically  sort-vars
  43:11  warning  Unexpected console statement                                                 no-console

/opt/numenta/imbu/gui/browser/components/search.jsx
  95:1  warning  Line 95 exceeds the maximum line length of 80  max-len

/opt/numenta/imbu/gui/browser/stores/search.js
  47:3  warning  Missing JSDoc @returns for function    valid-jsdoc
  54:3  warning  Missing JSDoc @returns for function    valid-jsdoc
  61:3  warning  Missing JSDoc @returns for function    valid-jsdoc
  68:3  warning  Missing JSDoc @returns for function    valid-jsdoc
  75:3  warning  Missing JSDoc for parameter 'payload'  valid-jsdoc

✖ 8 problems (0 errors, 8 warnings)


> Imbu@0.0.1 build /opt/numenta/imbu
> gulp webpack

[16:28:36] Requiring external module babel-core/register
[16:28:37] Using gulpfile /opt/numenta/imbu/gulpfile.babel.js
[16:28:37] Starting 'webpack'...
[16:28:40] Version: webpack 1.12.9
        Asset     Size  Chunks             Chunk Names
    bundle.js  1.79 MB       0  [emitted]  main
bundle.js.map  2.14 MB       0  [emitted]  main
[16:28:40] Finished 'webpack' after 3.49 s
 ---> 8638edcd1c08
Removing intermediate container 0af939449549
Step 16 : CMD ./start_imbu.sh
 ---> Running in 7e74e8d6e2db
 ---> c8d56911aaa0
Removing intermediate container 7e74e8d6e2db
Successfully built c8d56911aaa0

```

**Comment by oxtopus** [comment]
Hmmm.  I will try again.  Here's the full output starting at Step 12: https://gist.github.com/oxtopus/57b67708e33e0c38738e

**Comment by lscheinkman** [comment]
according to your log `eslint`  was installed in step 13 (https://gist.github.com/oxtopus/57b67708e33e0c38738e#file-gistfile1-txt-L184).
Not sure why it cannot find it later.

**Comment by BoltzmannBrain** [comment]
I'm having an issue with the npm build as well... my webpack is v1.12.2, not v1.12.9 (although 1.12.9 was installed in step 13).
My output is the same as Luiz's except for the following:

    > Imbu@0.0.1 build /opt/numenta/imbu
    > gulp webpack

    [15:38:55] Requiring external module babel-core/register
    [15:38:55] Using gulpfile /opt/numenta/imbu/gulpfile.babel.js
    [15:38:55] Starting 'webpack'...
    [15:38:59] Version: webpack 1.12.2
            Asset     Size  Chunks             Chunk Names
        bundle.js  1.62 MB       0  [emitted]  main
    bundle.js.map  1.94 MB       0  [emitted]  main

    ERROR in ./gui/browser/app.js
    Module not found: Error: Cannot resolve module 'react-dom' in /opt/numenta/imbu/gui/browser
     @ ./gui/browser/app.js 40:16-36

    ERROR in ./gui/browser/components/main.jsx
    Module not found: Error: Cannot resolve module 'material-ui/lib/styles/raw-themes/light-raw-theme' in /opt/numenta/imbu/gui/browser/components
     @ ./gui/browser/components/main.jsx 61:49-109

    ERROR in ./gui/browser/components/search.jsx
    Module not found: Error: Cannot resolve module 'react-dom' in /opt/numenta/imbu/gui/browser/components
     @ ./gui/browser/components/search.jsx 37:16-36
    [15:38:59] Finished 'webpack' after 4.53 s
     ---> 27640e3e806d
    Removing intermediate container d9ef63434788
    Step 16 : CMD ./start_imbu.sh
     ---> Running in 71ffdc31e58c
     ---> 39d1d3f6415f
    Removing intermediate container 71ffdc31e58c
    Successfully built 39d1d3f6415f

File: imbu/engine/fluent_api.py [hunk 1]
```diff
@@ -214,15 +369,25 @@ class FluentAPIHandler(object):
...
+
+    return json.dumps(g_ready)
+
+
   def POST(self, modelName=_DEFAULT_MODEL_NAME): # pylint: disable=R0201,C0103
     addStandardHeaders()
     addCORSHeaders()
 
-    if modelName not in g_fluent.models:
+    if modelName not in g_models:
```
Comment (lscheinkman) [inline imbu/engine/fluent_api.py]: Should `g_fluent.samples` be replaced with  `g_samples` on line https://github.com/numenta/numenta-apps/pull/509/files#diff-21289c3a6ff81b015652c7866a664149L240 ?

Comment (oxtopus) [inline imbu/engine/fluent_api.py]: Yep!  Done.  Thanks.

**Comment by lscheinkman** [comment]
@oxtopus :+1: LGTM
I will create a separate JIRA so we can track the different docker issues you and Alex were having.

**Comment by BoltzmannBrain** [comment]
Thank you @lscheinkman

**Pull request merged**